django/contrib/postgres/search.py

```diff
@@ -144,10 +144,7 @@ def as_sql(self, compiler, connection, function=None, template=None):
             weight_sql, extra_params = compiler.compile(clone.weight)
             sql = "setweight({}, {})".format(sql, weight_sql)
 
-        # These parameters must be bound on the client side because we may
-        # want to create an index on this expression.
-        sql = connection.ops.compose_sql(sql, config_params + params + extra_params)
-        return sql, []
```

Comment:
This was only necessary for a single test `postgres_tests.test_search.SearchVectorIndexTests.test_search_vector_index`, however `test_search_vector_index` didn't cover the way how Django handles creating new constraints. It was a temporary and bit clunky way to create a regression test.

Reply:
Ah in the end you found a nice result 👍